### ISSUE TITLE
Add templates for generated pages

### ISSUE THREAD

**Issue by melissawm** [melissawm/sphinx-tags#3]

**Comment by nebelgrau77** [comment]
How would that work? I still know very little about Sphinx, so I'm not sure how you intend this.

**Comment by melissawm** [comment]
I was thinking to use some sort of templating to configure the look and feel of the tags overview page. For example: https://stackoverflow.com/questions/13209597/override-html-page-template-for-a-specific-sphinx-document

 I tried looking for solutions but I don't have any specific ideas right now.

**Comment by nebelgrau77** [comment]
Yeah, I get the template idea, I have some minor modifications like that in my project (some are pretty difficult to implement, e.g. if you want your logo in the sidebar to be below the title - no problem! but to have it above it I ended up making a single logo+title png file :D)
But would you like to add some templates for the users to choose from, or somehow have the extension to generate them?

**Comment by JWCook** [comment]
I would find this useful as well.

Instead of making HTML templates, this could be done with 4 separate markup templates:
* Tags overview (rST)
* Tag page (rST)
* Tags overview (MyST)
* Tag page (MyST)

This would basically replace what sphinx-tags currently does in [`Tag.create_file()`](https://github.com/melissawm/sphinx-tags/blob/7338142188940c1d2772b56c105f74e9c1d82539/src/sphinx_tags/__init__.py#L148-L179) and [`Entry.tagpage()`](https://github.com/melissawm/sphinx-tags/blob/7338142188940c1d2772b56c105f74e9c1d82539/src/sphinx_tags/__init__.py#L231-L265).

Here's an example for the [autosummary](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html) extension, with both the directive code and template files: https://github.com/sphinx-doc/sphinx/tree/master/sphinx/ext/autosummary
* The part that renders the templates is in [AutosummaryRenderer](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/autosummary/generate.py#L114).
* The context passed to those templates is built in [generate_autosummary_content()](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/autosummary/generate.py#L225).

Here's a quick (probably incorrect, but close enough) example of what a MyST template for `tagsoverview.md` might look like:
````jinja
(tagoverview)=
# {{ tags_overview_title }}

{% raw %}
```{toctree}
{% endraw %}
---
caption: {{ tags_index_head }}
maxdepth: 1
---
{% for tag in tags %}
  {{ tag.name }} ({{ len(tag.items) }}) <{{ tag.name }}>
{%- endfor %}
```
````

> templating to configure the look and feel of the tags overview page

For customizing style instead of content, that can already be done on the user's end with Sphinx's [`html_css_files`](https://docs.readthedocs.io/en/stable/guides/adding-custom-css.html) option.
